apps/omg_watcher/lib/exit_processor/competitor_info.ex

```diff
@@ -49,16 +50,21 @@ defmodule OMG.Watcher.ExitProcessor.CompetitorInfo do
          }}
       )
       when is_integer(input_index) and is_binary(signature) do
-    value = %{tx: tx, competing_input_index: input_index, competing_input_signature: signature}
+    value = %{
+      tx: InFlightExitInfo.to_db_value(tx),
+      competing_input_index: input_index,
+      competing_input_signature: signature
+    }
+
     {:put, :competitor_info, {tx_hash, value}}
   end
 
-  def from_db_kv({tx_hash, %__MODULE__{} = competitor}), do: from_db_kv({tx_hash, Map.from_struct(competitor)})
-
   def from_db_kv(
-        {tx_hash, %{tx: %Transaction.Signed{} = tx, competing_input_index: index, competing_input_signature: signature}}
+        {tx_hash, %{tx: %{} = signed_tx_map, competing_input_index: index, competing_input_signature: signature}}
```

Comment:
why not add a guard `when is_map(signed_tx_map)`? I think this is more consistent

Reply:
I didn't, thinking it wouldn't match against a struct then, but it seems `is_map(struct) == true` so ok